### DISCUSSION TITLE
How to add external Pydevlake plugin?

### DISCUSSION THREAD

**Discussion by gffuentes** [apache/devlake#6930] [Q&A]
Hello there - our devlake implementation uses the helm chart and we are interested in creating some plugins using [Pydevlake](https://github.com/apache/incubator-devlake/blob/main/backend/python/README.md).

The instruction on how to create and develop one are pretty thorough, however i'm not sure of what the best way to actually install the plugin into devlake, especially if its running from the helm chart.

It looks like Poetry is used to create the new plugin as a python package. Does that mean the package can be hosted separately (in a private repo) and devlake can be pointed to go get it and install? or does the package HAVE to live in `python/plugins`.

Would love it if i can just create the plugin in a separate repo and just point devlake to it via helm chart config.

Thanks!

**Comment by klesh** [comment]
@keon94 @CamilleTeruel Would like your input for this. Thanks.

**Comment by mlagergren** [comment]
+1. I came here looking for exactly this. I'm looking into implement plugins using Pydevlake and ran into the same question (not using helm though).

**Comment by klesh** [comment]
It should be doable by mounting the plugin dir as a sub dir of `incubator-devlake/backend/python/plugins`

**Comment by klesh** [comment]
@ZhangNing10 @JorgeGar

**Comment by parraletz** [comment]
I'd like to know the same, how can I integrate a Pydevlake plugin?  :D
